src/storage/node-fs.ts

```diff
@@ -1,54 +1,71 @@
-import type { Stats } from "node:fs";
 import { resolve, parse, join } from "pathe";
 import { createError } from "h3";
 import { cachedPromise, getEnv } from "../utils";
 import type { IPXStorage } from "../types";
 
 export type NodeFSSOptions = {
-  dir?: string;
+  dir?: string | string[];
   maxAge?: number;
 };
 
+function resolveDirs(options: NodeFSSOptions) {
+  if (!options.dir || !Array.isArray(options.dir)) {
+    const dir = resolve(options.dir || getEnv("IPX_FS_DIR") || ".");
+    return [dir];
+  }
+
+  return options.dir.map((dir) => {
+    return resolve(dir);
+  });
+}
+
 export function ipxFSStorage(_options: NodeFSSOptions = {}): IPXStorage {
-  const rootDir = resolve(_options.dir || getEnv("IPX_FS_DIR") || ".");
+  const dirs = resolveDirs(_options);
   const maxAge = _options.maxAge || getEnv("IPX_FS_MAX_AGE");
 
-  const _resolve = (id: string) => {
-    const resolved = join(rootDir, id);
-    if (!isValidPath(resolved) || !resolved.startsWith(rootDir)) {
+  const _getFS = cachedPromise(() => import("node:fs/promises"));
+
+  const getMeta = async (id: string) => {
```

Comment:
Since this is reused also for reading, I think we can rename util to `resolveFile`